File: Modules/ServerTypes/hospital/DocSpecForm.ts
import { IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface DocSpecForm {
    SpecialityId: IntegerEditor;
    DoctorId: IntegerEditor;
}

export class DocSpecForm extends PrefixedContext {
    static readonly formKey = 'hospital.DocSpec';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!DocSpecForm.init)  {
            DocSpecForm.init = true;

            var w0 = IntegerEditor;

            initFormType(DocSpecForm, [
                'SpecialityId', w0,
                'DoctorId', w0
            ]);
        }
    }
}

File: Modules/hospital/DocSpec/DocSpecForm.cs
using Serenity.ComponentModel;

namespace Serene2.hospital.Forms;

[FormScript("hospital.DocSpec")]
[BasedOnRow(typeof(DocSpecRow), CheckNames = true)]
public class DocSpecForm
{
    public int SpecialityId { get; set; }
    public int DoctorId { get; set; }
}